### PR TITLE
[zstr] Fix include issues, add usage message override, add homepage and zlib dependency to vcpkg.json

### LINKED ISSUE #16402: [zstr] zstr_make_unique_polyfill.h missing, causing compile errors on C++11 projects
**Host Environment**
- OS: Xubuntu 20.04.2 LTS
- Compiler: Clang 10

**To Reproduce**
1. `[...]/vcpkg$ ./vcpkg install zstr`
2. Create these files:
- CMakeLists.txt:
```cmake
add_executable(foo src/foo.cpp)

set_target_properties(foo PROPERTIES
    CXX_STANDARD 11
    CXX_STANDARD_REQUIRED ON
)

find_package(ZLIB REQUIRED)
find_path(ZSTR_INCLUDE_DIRS "zstr/strict_fstream.hpp")

target_link_libraries(foo PRIVATE ZLIB::ZLIB)
target_include_directories(foo PRIVATE ${ZSTR_INCLUDE_DIRS})
```
- src/foo.cpp:
```cpp
#include <zstr/zstr.hpp>

int main()
    {return 0;}
```
3. `[...]/build$ cmake .. -DCMAKE_TOOLCHAIN_FILE:FILEPATH=[...]/vcpkg/scripts/buildsystems/vcpkg.cmake`
4. `[...]/build$ cmake --build .`

**Failure logs** 
```
In file included from [...]/src/foo.cpp:1:
[...]/vcpkg/installed/x64-linux/include/zstr/zstr.hpp:20:10: fatal error: zstr_make_unique_polyfill.h: No such file or directory
   20 | #include <zstr_make_unique_polyfill.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Additional context**
https://github.com/mateidavid/zstr/blob/v1.0.4/src/zstr.hpp#L19-L21

### LINKED ISSUE #18067: [zstr] zstr.hpp can't include strict_fstream.hpp due to being installed inside include/zstr
**Host Environment**
- OS: Windows 10
- Compiler: Clang 11

**To Reproduce**
1. `[...]/vcpkg$ ./vcpkg install zstr`
2. Create these files:
- CMakeLists.txt:
```cmake
add_executable(foo src/foo.cpp)

find_package(ZLIB REQUIRED)
find_path(ZSTR_INCLUDE_DIRS "zstr/strict_fstream.hpp")

target_link_libraries(foo PRIVATE ZLIB::ZLIB)
target_include_directories(foo PRIVATE ${ZSTR_INCLUDE_DIRS})
```
- src/foo.cpp:
```cpp
#include <zstr/zstr.hpp>

int main()
    {return 0;}
```
3. `[...]/build$ cmake .. -DCMAKE_TOOLCHAIN_FILE:FILEPATH=[...]/vcpkg/scripts/buildsystems/vcpkg.cmake`
4. `[...]/build$ cmake --build .`

**Failure logs** 
```
In file included from [...]/src/foo.cpp:1:
[...]/vcpkg/installed/x64-windows-static/include\zstr/zstr.hpp:15:10: error: 'strict_fstream.hpp' file not found with <angled> include; use "quotes" instead
#include <strict_fstream.hpp>
          ^
```

### PR DESCRIPTION
**Pull request by dan9er**
- #### What does your PR fix?  
  This PR fixes 2 include issues with the zstr port that were causing compile errors on projects that used it. It also adds an usage file, so users are directed to put the more sensible `find_path(ZSTR_INCLUDE_DIRS "zstr.hpp")` in their CMakeLists instead of `find_path(ZSTR_INCLUDE_DIRS "strict_fstream.hpp")`.
  Fixes #16402
  Fixes #18067

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  No, zstr is a header only library.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  I haven't yet, I'm making this as a draft PR. This is my first PR for vcpkg and i'm not sure whether my fix for #18067 is acceptable or not.

### CODE REVIEW

File: ports/zstr/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "zstr",
+  "version": "1.0.4",
+  "port-version": 1,
+  "description": "This C++ header-only library enables the use of C++ standard iostreams to access ZLib-compressed streams.",
+  "homepage": "https://github.com/mateidavid/zstr",
+  "dependencies": [ "zlib" ]
```
Comment (JackBoosY) [inline ports/zstr/vcpkg.json:7]:
```suggestion
  "dependencies": [
    "zlib"
  ]
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Please also run `./vcpkg x-add-version zstr` then commit changes.

File: ports/zstr/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "zstr",
+  "version": "1.0.4",
+  "port-version": 1,
+  "description": "This C++ header-only library enables the use of C++ standard iostreams to access ZLib-compressed streams.",
+  "homepage": "https://github.com/mateidavid/zstr",
+  "dependencies": [
+    "zlib"
```
Comment (JackBoosY) [inline ports/zstr/vcpkg.json:8]: From https://github.com/mateidavid/zstr#input-auto-detection:
Should we also add gzip and set gzip and zlib as features?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by dg0yt** [comment]
No matter where this gets installed: Please verify that the usage instructions make sense (`#include <zstr.hpp>`, cf. #18067).

File: ports/zstr/vcpkg.json (hunk 2 above)
Comment (dan9er) [reply ports/zstr/vcpkg.json:8]: 1. gzip uses zlib, it just wraps the resulting compressed stream diferently.
  https://stackoverflow.com/a/20765054
  https://dev.to/biellls/compression-clearing-the-confusion-on-zip-gzip-zlib-and-deflate-15g1
2. zstr does not have configure options.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Ready for review now? Did you test the usage?

**Comment by JackBoosY** [comment]
This PR has been inactive for a long time, please reopen it if there is any progress.

**Comment by dan9er** [comment]
Sorry for abandoning this PR! Life issues got in the way for a long while, but now I'm back. Can this please be reopened?

* I've already merged the current `microsoft:master` into `dan9er:zstr-fix` and resolved all conflicts.
* I can comfirm, if you make the required changes in a dependant project (`find_path(ZSTR_INCLUDE_DIRS "zstr/strict_fstream.hpp")` -> `find_path(ZSTR_INCLUDE_DIRS "strict_fstream.hpp")`) , `#include <zstr/zstr.hpp>` -> `#include <zstr.hpp>`) it will now compile on `x64-windows-static`. This is a compatability break, but the port was broken & unusable in the first place (see mentioned issues).
* **There is still the issue of vcpkg suggesting to find `strict_fstream.h` instead of `zstr.hpp`.** I believe it's because it just selects the first file alphabetically (which in my opinion is a false assumption), I would check myself but it's late where I am, so I'll check on Friday unless someone else does by then.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**